dipy/reconst/ivim.py

```diff
@@ -0,0 +1,397 @@
+
+            Where:
+            .. math::
+
+            S_0, f, D\* and D are the IVIM parameters.
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+            Gradient directions and bvalues
+
+        split_b : float, optional
+            The b-value to split the data on for two-stage fit.
+            default : 200.
+
+        bounds : tuple of arrays with 4 elements, optional
+            Bounds to constrain the fitted model parameters. This is only supported for
+            Scipy version > 0.17. When using a older scipy version, this function will raise
+            an error if bounds are different from None.
+            default : ([0., 0., 0., 0.], [np.inf, 1., 1., 1.])
+
+        tol : float, optional
+            Tolerance for convergence of minimization.
+            default : 1e-7
+
+        options : dict, optional
+            Dictionary containing gtol, ftol, eps and maxiter. This is passed
+            to leastsq.
+            default : options={'gtol': 1e-7, 'ftol': 1e-7, 'eps': 1e-7, 'maxiter': 1000}
+
+        References
+        ----------
+        .. [1] Le Bihan, Denis, et al. "Separation of diffusion
+                   and perfusion in intravoxel incoherent motion MR
+                   imaging." Radiology 168.2 (1988): 497-505.
+        .. [2] Federau, Christian, et al. "Quantitative measurement
+                   of brain perfusion with intravoxel incoherent motion
+                   MR imaging." Radiology 265.3 (2012): 874-881.
+        """
+        ReconstModel.__init__(self, gtab)
+        self.split_b = split_b
+        self.bounds = bounds
+        self.tol = tol
+        self.options = options
+
+        if SCIPY_LESS_0_17 and self.bounds is not None:
+            e_s = "Scipy versions less than 0.17 do not support "
+            e_s += "bounds. Please update to Scipy 0.17 to use bounds"
+            raise ValueError(e_s)
+        else:
+            self.bounds = (np.array([0., 0., 0., 0.]),
+                           np.array([np.inf, 1., 0.1, 0.1]))
+
+    def estimate_x0(self, data):
+        """
+        Fit the ivim params using a two stage fit.
+
+        In the two stage fitting routine, initially, we fit the signal
+        at bvals higher than the specified `split_b` using `TensorModel`
+        and get an intial guess for f and D. Then, using these parameters
+        we fit the entire data for all bvalues.
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel. A multi voxel decorator
+            will be applied to this fit method to scale it and apply it
+            to multiple voxels.
+
+        Returns
+        -------
+        x0_guess : array
+            An array with initial values of S0, f, D_star, D for each voxel.
+        """
+        S0_hat, D_guess = self._estimate_S0_D(data)
+        f_guess = 1 - S0_hat / np.mean(data[self.gtab.b0s_mask])
```

Comment:
It's true that data may not have an image at b=0. I think we could probably calculate the value at the lowest b-value without too much trouble. That's essentially what we're doing, but we're just assuming it's 0.